src/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs

```diff
@@ -319,7 +319,10 @@ public static RegistryKey OpenRemoteBaseKey(RegistryHive hKey, string machineNam
         /// <returns>the Subkey requested, or <b>null</b> if the operation failed.</returns>
         public RegistryKey OpenSubKey(string name, bool writable)
         {
-            return InternalOpenSubKey(name, writable);
+            ValidateKeyName(name);
+            EnsureNotDisposed();
+            name = FixupName(name); // Fixup multiple slashes to a single slash
+            return InternalOpenSubKeyCore(name, GetRegistryKeyRights(writable), throwOnPermissionFailure: true);
```

Comment:
Ah, this could be simplified to just:

```c#
public RegistryKey OpenSubKey(string name, bool writable)
{
    return OpenSubKey(name, GetRegistryKeyRights(writable));
}
```

The reason for the duplicated code with `OpenSubKey(string, RegistryRights)` is because I had originally kept `throwOnPermissionFailure: false` from inside `InternalOpenSubKey` below, but after investigating further, I noticed that it should be `throwOnPermissionFailure: true` and just made that modification without realizing the whole method now matched `OpenSubKey(string, RegistryRights)`'s implementation. I'll simplify.